File: typescript/difference-of-squares/difference-of-squares.ts
class Squares {

	squareOfSum: number; 
	sumOfSquares: number; 
	difference: number; 

	constructor(n: number) {
		this.squareOfSum = (n * (n+1) /2) ** 2;
		this.sumOfSquares = n * (n+1) * (2*n+1) / 6;
		this.difference = this.squareOfSum - this.sumOfSquares;
	}
}

export default Squares

File: scala/difference-of-squares/src/main/scala/DifferenceOfSquares.scala
object DifferenceOfSquares {

  def sumOfSquares(n: Int): Int = (1 to n).map(x => x*x).sum

  def squareOfSum(n: Int): Int = (1 to n).sum * (1 to n).sum

  def differenceOfSquares(n: Int): Int = squareOfSum(n) - sumOfSquares(n)
}
